cpp/oneapi/dal/algo/linear_kernel/backend/gpu/compute_kernel_dense_dpc.cpp

```diff
@@ -15,61 +15,54 @@
+static result_t compute(const context_gpu& ctx, const descriptor_t& desc, const input_t& input) {
+    const auto x = input.get_x();
+    const auto y = input.get_y();
 
-template <typename Float>
-static result_t call_daal_kernel(const context_gpu& ctx,
-                                 const descriptor_t& desc,
-                                 const table& x,
-                                 const table& y) {
     auto& queue = ctx.get_queue();
-    interop::execution_context_guard guard(queue);
 
     const int64_t row_count_x = x.get_row_count();
+    const int64_t col_count_x = x.get_column_count();
     const int64_t row_count_y = y.get_row_count();
+    const int64_t col_count_y = y.get_column_count();
 
+    ONEDAL_ASSERT(col_count_x == col_count_y);
     dal::detail::check_mul_overflow(row_count_x, row_count_y);
-    auto arr_values =
-        array<Float>::empty(queue, row_count_x * row_count_y, sycl::usm::alloc::device);
-
-    const auto daal_x = interop::convert_to_daal_table(queue, x);
-    const auto daal_y = interop::convert_to_daal_table(queue, y);
-    const auto daal_values =
-        interop::convert_to_daal_table(queue, arr_values, row_count_x, row_count_y);
-
-    daal_linear_kernel::Parameter daal_parameter(desc.get_scale(), desc.get_shift());
-    daal_linear_kernel_t<Float>().compute(daal_x.get(),
-                                          daal_y.get(),
-                                          daal_values.get(),
-                                          &daal_parameter);
-
-    return result_t{}.set_values(
-        dal::detail::homogen_table_builder{}.reset(arr_values, row_count_x, row_count_y).build());
-}
 
-template <typename Float>
-static result_t compute(const context_gpu& ctx, const descriptor_t& desc, const input_t& input) {
-    return call_daal_kernel<Float>(ctx, desc, input.get_x(), input.get_y());
+    const Float alpha = desc.get_scale();
+    const Float beta = desc.get_shift();
+
+    auto rows_block_x = row_accessor<const Float>(x).pull(queue, { 0, -1 });
+    const auto ndarray_x =
+        primitives::ndarray<Float, 2>::wrap(rows_block_x.get_data(), { row_count_x, col_count_x });
+
+    auto rows_block_y = row_accessor<const Float>(y).pull(queue, { 0, -1 });
+    const auto ndarray_y =
+        primitives::ndarray<Float, 2>::wrap(rows_block_y.get_data(), { row_count_y, col_count_y });
+
+    auto ndarray_res = primitives::ndarray<Float, 2>::empty(queue, { row_count_x, row_count_y });
+    sycl::event event_res;
+    if (beta != 0.0)
+        event_res = ndarray_res.fill(queue, Float(1));
+
+    gemm(queue, ndarray_x, ndarray_y.t(), ndarray_res, alpha, beta, { event_res }).wait_and_throw();
+
+    return result_t{}.set_values(dal::detail::homogen_table_builder{}
+                                     .reset(ndarray_res.flatten(queue), row_count_x, row_count_y)
+                                     .build());
```

Comment:
I'd suggest to not use builder for such simple cases. We can simplify this significantly if we slightly extend `homogen_table` by adding static `wrap` method.

```c++
const auto values = homogen_table::wrap(ndarray_res.flatten(queue), row_count_x, row_count_y);
return result_t{}.set_values(values );
```

There is already `wrap` for raw pointer in `homogen_table`. Similar method for `array` can be added.

Reply:
@vlad-nazarov, I've already implemented `wrap`, will create PR soon

Reply:
@rlnx, Added wrap method